include/sampling/ellipsoid.hpp

```diff
@@ -17,23 +17,71 @@
 template <typename Point>
 struct GetPointInDellipsoid
 {
-    template <typename NT, typename VT, typename MT, typename RandomNumberGenerator>
+    typedef typename Point::FT NT;
+
+    template <typename Ellipsoid, typename RandomNumberGenerator>
     inline static Point apply(unsigned int const& dim,
-                              VT const& eigenvalues_inv_sqrt, // sqrt of inverse of eigenvalues of matrix A in (x'Ax <= 1)
-                              MT const& EigenVectors,         // eigenvectors of matrix A in (x'Ax <= 1)
+                              Ellipsoid const& E,
                               RandomNumberGenerator& rng)
     {
         // Generate a point inside a sphere of radius 1.0
         Point p = GetPointInDsphere<Point>::apply(dim, NT(1.0), rng);
 
-        // scale points to the ellipsoid using the eigenvalues
-        VT scaled_vec = p.getCoefficients().cwiseProduct(eigenvalues_inv_sqrt);
+        // transform it to a point inside an ellipsoid
+        return Point(E.mult_Lcov(p.getCoefficients()));
+    }
+};
+
+
+template <typename Point>
+struct GetGaussianDirection
+{
+    typedef typename Point::FT NT;
+    typedef typename Eigen::Matrix<NT, Eigen::Dynamic, 1> VT;
+
+    template <typename Ellipsoid, typename RandomNumberGenerator>
+    inline static Point apply(unsigned int const& dim,
+                              Ellipsoid const& E,   // ellipsoid representing the Gaussian distribution
+                              RandomNumberGenerator &rng,
+                              bool normalized=false)
+    {
+        // Generate a point inside a sphere of radius 1.0
+        Point p = GetDirection<Point>::apply(dim ,rng, false);
+
+        // Multiply with cholesky matrix
+        VT gaussian_vec = E.mult_Lcov(p.getCoefficients());
 
-        // rotate with the eigenvectors
-        return Point(EigenVectors * scaled_vec);
+        if (normalized) {
+            gaussian_vec.normalize();
+        }
+
+        // convert to point
+        return Point(gaussian_vec);
     }
 };
 
 
+template <typename Point>
+struct GetPointOnDellipsoid
+{
+    typedef typename Point::FT NT;
+    typedef typename Eigen::Matrix<NT, Eigen::Dynamic, 1> VT;
+
+    template <typename Ellipsoid, typename RandomNumberGenerator>
+    inline static Point apply(unsigned int const& dim,
+                              Ellipsoid const& E,   // ellipsoid representing the Gaussian distribution
+                              RandomNumberGenerator &rng)
+    {
+        // Generate a point inside a sphere of radius 1.0
+        Point p = GetDirection<Point>::apply(dim ,rng, true);
```

Comment:
Plz fix the comma [i.e. ..(dim, rng, true)]

Reply:
fixed 👍🏽